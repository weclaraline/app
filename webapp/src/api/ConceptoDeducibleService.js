import ServiceAPI from "./ServiceAPI"

class ConceptoDeducibleService extends ServiceAPI {
    
    constructor(ConceptoDeducibleRequest){
        super();
        this.request = ConceptoDeducibleRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest()
                        .get("/recommendations",this.request);
        return data;
    }
}

export default ConceptoDeducibleService;