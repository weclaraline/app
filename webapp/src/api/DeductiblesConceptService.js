import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";
class DeductiblesConceptService extends ServiceAPI {
    
    constructor(ConceptoDeducibleRequest){
        super();
        this.request = ConceptoDeducibleRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get("/recommendations",this.request);
        return data;
    }
}

export default DeductiblesConceptService;