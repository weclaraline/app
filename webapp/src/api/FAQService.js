import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class FAQService extends ServiceAPI {
    
    constructor(FAQRequest){
        super();
        this.request = FAQRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get("/faq",this.request);
        return data;
    }
}

export default FAQService;