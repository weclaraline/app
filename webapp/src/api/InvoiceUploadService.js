import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class InvoiceUploadService extends ServiceAPI {
    
    constructor(InvoiceUploadRequest){
        super();
        this.request = InvoiceUploadRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .post("invoices/upload",this.request);
        return data;
    }
}

export default InvoiceUploadService;