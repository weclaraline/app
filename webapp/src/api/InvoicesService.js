import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class InvoicesService extends ServiceAPI {
    
    constructor(InvoicesRequest){
        super();
        this.request = InvoicesRequest;
    }

    async make(month, year){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get(`invoices/${month}/${year}`,this.request);
        return data;
    }
}

export default InvoicesService;