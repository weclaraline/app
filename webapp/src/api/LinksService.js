import ServiceAPI from "./ServiceAPI"
import { getCurrentLoggedUserGoogleId } from "../utils/LogIn";

class LinksService extends ServiceAPI {
    
    constructor(LinksRequest){
        super();
        this.request = LinksRequest;
    }

    async make(){
        const { data } = await this
                        .createRequest( getCurrentLoggedUserGoogleId() )
                        .get("/links",this.request);
        return data;
    }
}

export default LinksService;