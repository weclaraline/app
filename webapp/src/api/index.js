export { default as ServiceAPI } from './ServiceAPI'
export { default as StatusCodes } from './StatusCodes'
export { default as ConceptoDeducibleService } from './ConceptoDeducibleService'