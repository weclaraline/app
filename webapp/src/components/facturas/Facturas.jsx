import { Grid, makeStyles, Paper, Typography } from "@material-ui/core";
import React from "react";
import ConceptoDeduciblesComponent from "../facturas/ConceptoDeduciblesComponent";
import FAQEnlacesComponent from "../facturas/FAQEnlacesComponent";

const useStyles = makeStyles( (theme) => {
    const { main } = theme.palette.primary;
    const { overline, h1 } = theme.typography;
    return (
        {
            headerRow: {
                height: "200px",
                background: main,
                display: "flex",
                alignItems: "center",
                justifyContent: "center"
            },
            paper: {
                alignContent: "center",
                justifyContent: "center",
                color: "white",
                background: "transparent"
            },
            header: {
                h1
            },
            overline: {
                overline,
            },
        }
    )
});

const Facturas = () => {

    const classes = useStyles();

    return (
        <Grid container spacing={3}>
            <Grid 
                data-testid="title_banner" 
                item 
                className={classes.headerRow}
                xs={12}
            >
                <Paper elevation={0} className={classes.paper}>
                    <Typography variant={"h4"}>
                        Guía para solicitar facturas.
                    </Typography>
                    <Typography variant={"h6"}>
                        Sigue los pasos para que fácilmente...
                    </Typography>
                </Paper>
            </Grid>
            <Grid item xs={8}>
                <ConceptoDeduciblesComponent />
            </Grid>
            <Grid item xs={4}>
                <FAQEnlacesComponent />
            </Grid>    
        </Grid>
    );
}

export default Facturas;