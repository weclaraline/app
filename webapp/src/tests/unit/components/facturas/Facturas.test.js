import "../../testing_utils";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Facturas from "../../../../components/facturas/Facturas";
import { getByTestId, waitFor } from "@testing-library/react";
import { unmountComponentAtNode, render } from "react-dom";
import { ServiceAPI, StatusCodes } from "../../../../api/";
import nock from "nock";
import faker from "faker";

let container = null;

const build = () => render(<Facturas />, container);

const generateRecommendations = (quantity) => {
    let recommendations = []
    for( let i = 0; i < quantity; i++) {
        recommendations.push({
            description: faker.lorem.words(4),
        })
    }
    return recommendations;
}
const generateDeductibleTypesAndRecommendations = (quantity) => {
    let deductionTypes= [];
    for( let i = 0; i < quantity; i++){
        deductionTypes.push({
            key: faker.random.uuid,
            tipo_deduccion: faker.lorem.words(2),
            recomendaciones: generateRecommendations(Math.floor(Math.random * 5) + 1 ),
        })
    }
    return deductionTypes;
};
const recommendationsSuccessResponse =  generateDeductibleTypesAndRecommendations(4)
const createRecommendationsInterceptor = () => {
    nock(new ServiceAPI().getBaseURL())
    .get("/recomendaciones")
    .reply(StatusCodes.OK, recommendationsSuccessResponse);
}

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});
  
afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
    return container;
});

it("Renders original content", ()=> {
    const tree = shallow(<Facturas />);
    expect(toJson(tree)).toMatchSnapshot();
});

it("Calls recommendation service and loads recommendations", async () => {
    createRecommendationsInterceptor();
    build();
    await waitFor(() => {
        expect( getByTestId(container, "lista_tipo_deducibles") ).toBeInTheDocument();
        expect( getByTestId(container, "panel_faq_enlaces") ).toBeInTheDocument();
    });
});