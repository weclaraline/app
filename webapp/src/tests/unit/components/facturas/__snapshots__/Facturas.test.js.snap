// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders original content 1`] = `
<WithStyles(ForwardRef(Container))>
  <TipoDeduciblesComponent />
  <FAQEnlacesComponent />
</WithStyles(ForwardRef(Container))>
`;
